compiler/rustc_borrowck/src/diagnostics/mod.rs

```diff
@@ -1131,13 +1134,19 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                                 place_name, partially_str, loop_message
                             ),
                         );
-                        let ty = tcx.erase_regions(moved_place.ty(self.body, self.infcx.tcx).ty);
+                        let infcx = tcx.infer_ctxt().build();
+                        let ty = infcx.freshen(moved_place.ty(self.body, tcx).ty);
```

Comment:
I think it's slightly clearer here to just call `tcx.erase_regions` instead of freshening here.

Reply:
Erase regions still causes ICEs in the newly created infcx.

Reply:
```
---- [ui] src/test/ui/moves/pin-mut-reborrow.rs stdout ----

error: Error: expected failure status (Some(1)) but received status Some(101).
status: exit status: 101
command: "/Users/ekuber/workspace/rust/build/x86_64-apple-darwin/stage1/bin/rustc" "/Users/ekuber/workspace/rust/src/test/ui/moves/pin-mut-reborrow.rs" "-Zthreads=1" "--target=x86_64-apple-darwin" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/Users/ekuber/workspace/rust/build/x86_64-apple-darwin/test/ui/moves/pin-mut-reborrow" "-A" "unused" "-Crpath" "-Cdebuginfo=0" "-Lnative=/Users/ekuber/workspace/rust/build/x86_64-apple-darwin/native/rust-test-helpers" "-L" "/Users/ekuber/workspace/rust/build/x86_64-apple-darwin/test/ui/moves/pin-mut-reborrow/auxiliary"
stdout: none
--- stderr -------------------------------
thread 'rustc' panicked at 'index out of bounds: the len is 1 but the index is 4', /Users/ekuber/.cargo/registry/src/github.com-1ecc6299db9ec823/ena-0.14.0/src/snapshot_vec.rs:199:10
stack backtrace:
   0:        0x10de61661 - std::backtrace_rs::backtrace::libunwind::trace::h4c37c7b868f7da90
                               at /Users/ekuber/workspace/rust/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:        0x10de61661 - std::backtrace_rs::backtrace::trace_unsynchronized::h2f0b0601501a7c92
                               at /Users/ekuber/workspace/rust/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:        0x10de61661 - std::sys_common::backtrace::_print_fmt::hee48b30ee356c1de
                               at /Users/ekuber/workspace/rust/library/std/src/sys_common/backtrace.rs:65:5
   3:        0x10de61661 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h01a9852bec4a0f86
                               at /Users/ekuber/workspace/rust/library/std/src/sys_common/backtrace.rs:44:22
   4:        0x10def3bfa - core::fmt::write::hee388f67573ae7f4
                               at /Users/ekuber/workspace/rust/library/core/src/fmt/mod.rs:1208:17
   5:        0x10de46248 - std::io::Write::write_fmt::hf5559cb02408d32c
                               at /Users/ekuber/workspace/rust/library/std/src/io/mod.rs:1682:15
   6:        0x10de61482 - std::sys_common::backtrace::_print::h0c4f55672e7bf700
                               at /Users/ekuber/workspace/rust/library/std/src/sys_common/backtrace.rs:47:5
   7:        0x10de61482 - std::sys_common::backtrace::print::h4e51cef4588dceab
                               at /Users/ekuber/workspace/rust/library/std/src/sys_common/backtrace.rs:34:9
   8:        0x10de707c5 - std::panicking::default_hook::{{closure}}::h5461b55c8bd9353c
   9:        0x10de70600 - std::panicking::default_hook::h71169fa271a6712f
                               at /Users/ekuber/workspace/rust/library/std/src/panicking.rs:286:9
  10:        0x11d544c5d - <alloc[8206df779e47f41]::boxed::Box<dyn for<'a, 'b> core[3d5a77c8bca55160]::ops::function::Fn<(&'a core[3d5a77c8bca55160]::panic::panic_info::PanicInfo<'b>,), Output = ()> + core[3d5a77c8bca55160]::marker::Send + core[3d5a77c8bca55160]::marker::Sync> as core[3d5a77c8bca55160]::ops::function::Fn<(&core[3d5a77c8bca55160]::panic::panic_info::PanicInfo,)>>::call
                               at /Users/ekuber/workspace/rust/library/alloc/src/boxed.rs:2024:9
  11:        0x11d544c5d - rustc_driver[c86f44892fce95b7]::DEFAULT_HOOK::{closure#0}::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_driver/src/lib.rs:1202:13
  12:        0x10de70e3b - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::hcb3ecff25593f957
                               at /Users/ekuber/workspace/rust/library/alloc/src/boxed.rs:2024:9
  13:        0x10de70e3b - std::panicking::rust_panic_with_hook::h58dad9bc83cf7f8d
                               at /Users/ekuber/workspace/rust/library/std/src/panicking.rs:692:13
  14:        0x10de5a083 - std::panicking::begin_panic_handler::{{closure}}::h374f566aa199a332
                               at /Users/ekuber/workspace/rust/library/std/src/panicking.rs:579:13
  15:        0x10de59fb8 - std::sys_common::backtrace::__rust_end_short_backtrace::hb12d8d6a9e5c9eeb
                               at /Users/ekuber/workspace/rust/library/std/src/sys_common/backtrace.rs:137:18
  16:        0x10de70898 - rust_begin_unwind
                               at /Users/ekuber/workspace/rust/library/std/src/panicking.rs:575:5
  17:        0x10df0c5e3 - core::panicking::panic_fmt::h925f269d1227e413
                               at /Users/ekuber/workspace/rust/library/core/src/panicking.rs:64:14
  18:        0x10df0c726 - core::panicking::panic_bounds_check::h2175263d2dfb8f91
                               at /Users/ekuber/workspace/rust/library/core/src/panicking.rs:147:5
  19:        0x121c4666e - <ena[d0da655f2ec78c62]::snapshot_vec::SnapshotVec<ena[d0da655f2ec78c62]::unify::backing_vec::Delegate<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey>, &mut alloc[8206df779e47f41]::vec::Vec<ena[d0da655f2ec78c62]::unify::VarValue<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey>>, &mut rustc_infer[ac99d04a9fad1469]::infer::undo_log::InferCtxtUndoLogs>>::get
                               at /Users/ekuber/.cargo/registry/src/github.com-1ecc6299db9ec823/ena-0.14.0/src/snapshot_vec.rs:199:10
  20:        0x121c4666e - <ena[d0da655f2ec78c62]::snapshot_vec::SnapshotVec<ena[d0da655f2ec78c62]::unify::backing_vec::Delegate<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey>, &mut alloc[8206df779e47f41]::vec::Vec<ena[d0da655f2ec78c62]::unify::VarValue<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey>>, &mut rustc_infer[ac99d04a9fad1469]::infer::undo_log::InferCtxtUndoLogs> as core[3d5a77c8bca55160]::ops::index::Index<usize>>::index
                               at /Users/ekuber/.cargo/registry/src/github.com-1ecc6299db9ec823/ena-0.14.0/src/snapshot_vec.rs:319:9
  21:        0x121c6ac27 - <ena[d0da655f2ec78c62]::unify::backing_vec::InPlace<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey, &mut alloc[8206df779e47f41]::vec::Vec<ena[d0da655f2ec78c62]::unify::VarValue<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey>>, &mut rustc_infer[ac99d04a9fad1469]::infer::undo_log::InferCtxtUndoLogs> as core[3d5a77c8bca55160]::ops::index::Index<usize>>::index
                               at /Users/ekuber/.cargo/registry/src/github.com-1ecc6299db9ec823/ena-0.14.0/src/unify/backing_vec.rs:152:10
  22:        0x121c6ac27 - <ena[d0da655f2ec78c62]::unify::UnificationTable<ena[d0da655f2ec78c62]::unify::backing_vec::InPlace<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey, &mut alloc[8206df779e47f41]::vec::Vec<ena[d0da655f2ec78c62]::unify::VarValue<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey>>, &mut rustc_infer[ac99d04a9fad1469]::infer::undo_log::InferCtxtUndoLogs>>>::value
                               at /Users/ekuber/.cargo/registry/src/github.com-1ecc6299db9ec823/ena-0.14.0/src/unify/mod.rs:347:10
  23:        0x121c6ac27 - <ena[d0da655f2ec78c62]::unify::UnificationTable<ena[d0da655f2ec78c62]::unify::backing_vec::InPlace<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey, &mut alloc[8206df779e47f41]::vec::Vec<ena[d0da655f2ec78c62]::unify::VarValue<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey>>, &mut rustc_infer[ac99d04a9fad1469]::infer::undo_log::InferCtxtUndoLogs>>>::inlined_get_root_key
                               at /Users/ekuber/.cargo/registry/src/github.com-1ecc6299db9ec823/ena-0.14.0/src/unify/mod.rs:362:19
  24:        0x121c6ac27 - <ena[d0da655f2ec78c62]::unify::UnificationTable<ena[d0da655f2ec78c62]::unify::backing_vec::InPlace<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey, &mut alloc[8206df779e47f41]::vec::Vec<ena[d0da655f2ec78c62]::unify::VarValue<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey>>, &mut rustc_infer[ac99d04a9fad1469]::infer::undo_log::InferCtxtUndoLogs>>>::uninlined_get_root_key
                               at /Users/ekuber/.cargo/registry/src/github.com-1ecc6299db9ec823/ena-0.14.0/src/unify/mod.rs:381:9
  25:        0x121c69805 - <ena[d0da655f2ec78c62]::unify::UnificationTable<ena[d0da655f2ec78c62]::unify::backing_vec::InPlace<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey, &mut alloc[8206df779e47f41]::vec::Vec<ena[d0da655f2ec78c62]::unify::VarValue<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey>>, &mut rustc_infer[ac99d04a9fad1469]::infer::undo_log::InferCtxtUndoLogs>>>::unify_var_var::<rustc_middle[73d8f7044761bf4e]::ty::sty::RegionVid, rustc_middle[73d8f7044761bf4e]::ty::sty::RegionVid>
                               at /Users/ekuber/.cargo/registry/src/github.com-1ecc6299db9ec823/ena-0.14.0/src/unify/mod.rs:525:22
  26:        0x121c69805 - <ena[d0da655f2ec78c62]::unify::UnificationTable<ena[d0da655f2ec78c62]::unify::backing_vec::InPlace<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey, &mut alloc[8206df779e47f41]::vec::Vec<ena[d0da655f2ec78c62]::unify::VarValue<rustc_middle[73d8f7044761bf4e]::infer::unify_key::RegionVidKey>>, &mut rustc_infer[ac99d04a9fad1469]::infer::undo_log::InferCtxtUndoLogs>>>::union::<rustc_middle[73d8f7044761bf4e]::ty::sty::RegionVid, rustc_middle[73d8f7044761bf4e]::ty::sty::RegionVid>
                               at /Users/ekuber/.cargo/registry/src/github.com-1ecc6299db9ec823/ena-0.14.0/src/unify/mod.rs:482:9
  27:        0x121bbf649 - <rustc_infer[ac99d04a9fad1469]::infer::region_constraints::RegionConstraintCollector>::make_eqregion
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/region_constraints/mod.rs:522:21
  28:        0x121b745ab - <rustc_infer[ac99d04a9fad1469]::infer::equate::Equate as rustc_middle[73d8f7044761bf4e]::ty::relate::TypeRelation>::regions
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/equate.rs:160:9
  29:        0x121bc861c - <rustc_middle[73d8f7044761bf4e]::ty::sty::Region as rustc_middle[73d8f7044761bf4e]::ty::relate::Relate>::relate::<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/relate.rs:786:9
  30:        0x121bc861c - <rustc_infer[ac99d04a9fad1469]::infer::equate::Equate as rustc_middle[73d8f7044761bf4e]::ty::relate::TypeRelation>::relate::<rustc_middle[73d8f7044761bf4e]::ty::sty::Region>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/relate.rs:49:9
  31:        0x121bc861c - <rustc_infer[ac99d04a9fad1469]::infer::equate::Equate as rustc_middle[73d8f7044761bf4e]::ty::relate::TypeRelation>::relate_with_variance::<rustc_middle[73d8f7044761bf4e]::ty::sty::Region>
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/equate.rs:74:9
  32:        0x121bc861c - rustc_middle[73d8f7044761bf4e]::ty::relate::super_relate_tys::<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/relate.rs:490:21
  33:        0x121c61296 - <rustc_infer[ac99d04a9fad1469]::infer::InferCtxt>::super_combine_tys::<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/combine.rs:122:18
  34:        0x121b74dbc - <rustc_infer[ac99d04a9fad1469]::infer::equate::Equate as rustc_middle[73d8f7044761bf4e]::ty::relate::TypeRelation>::tys
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/equate.rs:146:17
  35:        0x121bb1e54 - <rustc_middle[73d8f7044761bf4e]::ty::Ty as rustc_middle[73d8f7044761bf4e]::ty::relate::Relate>::relate::<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/relate.rs:393:9
  36:        0x121bb1e54 - <rustc_infer[ac99d04a9fad1469]::infer::equate::Equate as rustc_middle[73d8f7044761bf4e]::ty::relate::TypeRelation>::relate::<rustc_middle[73d8f7044761bf4e]::ty::Ty>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/relate.rs:49:9
  37:        0x121bb1e54 - <rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg as rustc_middle[73d8f7044761bf4e]::ty::relate::Relate>::relate::<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/relate.rs:821:20
  38:        0x121bf97f3 - <rustc_infer[ac99d04a9fad1469]::infer::equate::Equate as rustc_middle[73d8f7044761bf4e]::ty::relate::TypeRelation>::relate::<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/relate.rs:49:9
  39:        0x121bf97f3 - <rustc_infer[ac99d04a9fad1469]::infer::equate::Equate as rustc_middle[73d8f7044761bf4e]::ty::relate::TypeRelation>::relate_with_variance::<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/equate.rs:74:9
  40:        0x121bf97f3 - rustc_middle[73d8f7044761bf4e]::ty::relate::relate_substs::<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/relate.rs:149:9
  41:        0x121bf97f3 - <&mut rustc_middle[73d8f7044761bf4e]::ty::relate::relate_substs<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>::{closure#0} as core[3d5a77c8bca55160]::ops::function::FnOnce<((rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg, rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg),)>>::call_once
                               at /Users/ekuber/workspace/rust/library/core/src/ops/function.rs:566:13
  42:        0x121bf97f3 - <core[3d5a77c8bca55160]::option::Option<(rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg, rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg)>>::map::<core[3d5a77c8bca55160]::result::Result<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg, rustc_middle[73d8f7044761bf4e]::ty::error::TypeError>, &mut rustc_middle[73d8f7044761bf4e]::ty::relate::relate_substs<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>::{closure#0}>
                               at /Users/ekuber/workspace/rust/library/core/src/option.rs:969:29
  43:        0x121bf97f3 - <core[3d5a77c8bca55160]::iter::adapters::map::Map<core[3d5a77c8bca55160]::iter::adapters::zip::Zip<core[3d5a77c8bca55160]::iter::adapters::copied::Copied<core[3d5a77c8bca55160]::slice::iter::Iter<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>>, core[3d5a77c8bca55160]::iter::adapters::copied::Copied<core[3d5a77c8bca55160]::slice::iter::Iter<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>>>, rustc_middle[73d8f7044761bf4e]::ty::relate::relate_substs<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>::{closure#0}> as core[3d5a77c8bca55160]::iter::traits::iterator::Iterator>::next
                               at /Users/ekuber/workspace/rust/library/core/src/iter/adapters/map.rs:103:9
  44:        0x121bf97f3 - <core[3d5a77c8bca55160]::result::Result<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg, rustc_middle[73d8f7044761bf4e]::ty::error::TypeError> as rustc_type_ir[80fbade2e7c847c0]::InternIteratorElement<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg, &rustc_middle[73d8f7044761bf4e]::ty::list::List<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>>>::intern_with::<core[3d5a77c8bca55160]::iter::adapters::map::Map<core[3d5a77c8bca55160]::iter::adapters::zip::Zip<core[3d5a77c8bca55160]::iter::adapters::copied::Copied<core[3d5a77c8bca55160]::slice::iter::Iter<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>>, core[3d5a77c8bca55160]::iter::adapters::copied::Copied<core[3d5a77c8bca55160]::slice::iter::Iter<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>>>, rustc_middle[73d8f7044761bf4e]::ty::relate::relate_substs<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>::{closure#0}>, <rustc_middle[73d8f7044761bf4e]::ty::context::TyCtxt>::mk_substs<core[3d5a77c8bca55160]::iter::adapters::map::Map<core[3d5a77c8bca55160]::iter::adapters::zip::Zip<core[3d5a77c8bca55160]::iter::adapters::copied::Copied<core[3d5a77c8bca55160]::slice::iter::Iter<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>>, core[3d5a77c8bca55160]::iter::adapters::copied::Copied<core[3d5a77c8bca55160]::slice::iter::Iter<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>>>, rustc_middle[73d8f7044761bf4e]::ty::relate::relate_substs<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>::{closure#0}>>::{closure#0}>
                               at /Users/ekuber/workspace/rust/compiler/rustc_type_ir/src/lib.rs:148:26
  45:        0x121b7429a - <core[3d5a77c8bca55160]::iter::adapters::map::Map<core[3d5a77c8bca55160]::iter::adapters::zip::Zip<core[3d5a77c8bca55160]::iter::adapters::copied::Copied<core[3d5a77c8bca55160]::slice::iter::Iter<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>>, core[3d5a77c8bca55160]::iter::adapters::copied::Copied<core[3d5a77c8bca55160]::slice::iter::Iter<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>>>, rustc_middle[73d8f7044761bf4e]::ty::relate::relate_substs<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>::{closure#0}> as rustc_type_ir[80fbade2e7c847c0]::InternAs<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg, &rustc_middle[73d8f7044761bf4e]::ty::list::List<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>>>::intern_with::<<rustc_middle[73d8f7044761bf4e]::ty::context::TyCtxt>::mk_substs<core[3d5a77c8bca55160]::iter::adapters::map::Map<core[3d5a77c8bca55160]::iter::adapters::zip::Zip<core[3d5a77c8bca55160]::iter::adapters::copied::Copied<core[3d5a77c8bca55160]::slice::iter::Iter<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>>, core[3d5a77c8bca55160]::iter::adapters::copied::Copied<core[3d5a77c8bca55160]::slice::iter::Iter<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>>>, rustc_middle[73d8f7044761bf4e]::ty::relate::relate_substs<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>::{closure#0}>>::{closure#0}>
                               at /Users/ekuber/workspace/rust/compiler/rustc_type_ir/src/lib.rs:78:9
  46:        0x121b7429a - <rustc_middle[73d8f7044761bf4e]::ty::context::TyCtxt>::mk_substs::<core[3d5a77c8bca55160]::iter::adapters::map::Map<core[3d5a77c8bca55160]::iter::adapters::zip::Zip<core[3d5a77c8bca55160]::iter::adapters::copied::Copied<core[3d5a77c8bca55160]::slice::iter::Iter<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>>, core[3d5a77c8bca55160]::iter::adapters::copied::Copied<core[3d5a77c8bca55160]::slice::iter::Iter<rustc_middle[73d8f7044761bf4e]::ty::subst::GenericArg>>>, rustc_middle[73d8f7044761bf4e]::ty::relate::relate_substs<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>::{closure#0}>>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:2854:9
  47:        0x121b7429a - rustc_middle[73d8f7044761bf4e]::ty::relate::relate_substs::<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/relate.rs:148:5
  48:        0x121b7429a - <rustc_infer[ac99d04a9fad1469]::infer::equate::Equate as rustc_middle[73d8f7044761bf4e]::ty::relate::TypeRelation>::relate_item_substs
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/equate.rs:64:9
  49:        0x121bc87a8 - rustc_middle[73d8f7044761bf4e]::ty::relate::super_relate_tys::<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/relate.rs:436:26
  50:        0x121c61296 - <rustc_infer[ac99d04a9fad1469]::infer::InferCtxt>::super_combine_tys::<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/combine.rs:122:18
  51:        0x121b74dbc - <rustc_infer[ac99d04a9fad1469]::infer::equate::Equate as rustc_middle[73d8f7044761bf4e]::ty::relate::TypeRelation>::tys
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/equate.rs:146:17
  52:        0x12077c26a - <rustc_middle[73d8f7044761bf4e]::ty::Ty as rustc_middle[73d8f7044761bf4e]::ty::relate::Relate>::relate::<rustc_infer[ac99d04a9fad1469]::infer::equate::Equate>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/relate.rs:393:9
  53:        0x12077c26a - <rustc_infer[ac99d04a9fad1469]::infer::equate::Equate as rustc_middle[73d8f7044761bf4e]::ty::relate::TypeRelation>::relate::<rustc_middle[73d8f7044761bf4e]::ty::Ty>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/relate.rs:49:9
  54:        0x12077c26a - <rustc_infer[ac99d04a9fad1469]::infer::at::Trace>::eq::<rustc_middle[73d8f7044761bf4e]::ty::Ty>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/at.rs:269:13
  55:        0x12077c26a - <rustc_infer[ac99d04a9fad1469]::infer::InferCtxt>::commit_if_ok::<rustc_infer[ac99d04a9fad1469]::infer::InferOk<()>, rustc_middle[73d8f7044761bf4e]::ty::error::TypeError, <rustc_infer[ac99d04a9fad1469]::infer::at::Trace>::eq<rustc_middle[73d8f7044761bf4e]::ty::Ty>::{closure#0}>
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/mod.rs:818:17
  56:        0x120759085 - <rustc_infer[ac99d04a9fad1469]::infer::at::Trace>::eq::<rustc_middle[73d8f7044761bf4e]::ty::Ty>
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/at.rs:267:9
  57:        0x120759085 - <rustc_infer[ac99d04a9fad1469]::infer::at::At>::eq::<rustc_middle[73d8f7044761bf4e]::ty::Ty>
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/at.rs:169:9
  58:        0x12077d6a9 - <rustc_infer[ac99d04a9fad1469]::infer::InferCtxt>::can_eq::<rustc_middle[73d8f7044761bf4e]::ty::Ty>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/mod.rs:902:13
  59:        0x12077d6a9 - <rustc_infer[ac99d04a9fad1469]::infer::InferCtxt>::probe::<core[3d5a77c8bca55160]::result::Result<(), rustc_middle[73d8f7044761bf4e]::ty::error::TypeError>, <rustc_infer[ac99d04a9fad1469]::infer::InferCtxt>::can_eq<rustc_middle[73d8f7044761bf4e]::ty::Ty>::{closure#0}>
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/mod.rs:838:17
  60:        0x12077d6a9 - <rustc_infer[ac99d04a9fad1469]::infer::InferCtxt>::can_eq::<rustc_middle[73d8f7044761bf4e]::ty::Ty>
                               at /Users/ekuber/workspace/rust/compiler/rustc_infer/src/infer/mod.rs:901:9
  61:        0x1209a3fe1 - <rustc_borrowck[873188a397b846b5]::MirBorrowckCtxt>::explain_captures
                               at /Users/ekuber/workspace/rust/compiler/rustc_borrowck/src/diagnostics/mod.rs:1147:32
  62:        0x120985772 - <rustc_borrowck[873188a397b846b5]::MirBorrowckCtxt>::report_use_of_moved_or_uninitialized
                               at /Users/ekuber/workspace/rust/compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs:206:21
  63:        0x1209aa36e - <rustc_borrowck[873188a397b846b5]::MirBorrowckCtxt>::check_if_full_path_is_moved
                               at /Users/ekuber/workspace/rust/compiler/rustc_borrowck/src/lib.rs:1625:13
  64:        0x1209aa48a - <rustc_borrowck[873188a397b846b5]::MirBorrowckCtxt>::check_if_path_or_subpath_is_moved
                               at /Users/ekuber/workspace/rust/compiler/rustc_borrowck/src/lib.rs:1707:9
  65:        0x1209a9f88 - <rustc_borrowck[873188a397b846b5]::MirBorrowckCtxt>::consume_operand
  66:        0x1209a507d - <rustc_borrowck[873188a397b846b5]::MirBorrowckCtxt>::consume_rvalue
  67:        0x1209a507d - <rustc_borrowck[873188a397b846b5]::MirBorrowckCtxt as rustc_mir_dataflow[99c12b449ff177c0]::framework::visitor::ResultsVisitor>::visit_statement_before_primary_effect
                               at /Users/ekuber/workspace/rust/compiler/rustc_borrowck/src/lib.rs:582:17
  68:        0x12075ba1d - <rustc_mir_dataflow[99c12b449ff177c0]::framework::direction::Forward as rustc_mir_dataflow[99c12b449ff177c0]::framework::direction::Direction>::visit_results_in_block::<rustc_borrowck[873188a397b846b5]::dataflow::BorrowckAnalyses<rustc_index[845bb2ac091a64a7]::bit_set::BitSet<rustc_borrowck[873188a397b846b5]::dataflow::BorrowIndex>, rustc_index[845bb2ac091a64a7]::bit_set::ChunkedBitSet<rustc_mir_dataflow[99c12b449ff177c0]::move_paths::MovePathIndex>, rustc_index[845bb2ac091a64a7]::bit_set::ChunkedBitSet<rustc_mir_dataflow[99c12b449ff177c0]::move_paths::InitIndex>>, rustc_borrowck[873188a397b846b5]::dataflow::BorrowckAnalyses<rustc_mir_dataflow[99c12b449ff177c0]::framework::engine::Results<rustc_borrowck[873188a397b846b5]::dataflow::Borrows>, rustc_mir_dataflow[99c12b449ff177c0]::framework::engine::Results<rustc_mir_dataflow[99c12b449ff177c0]::impls::MaybeUninitializedPlaces>, rustc_mir_dataflow[99c12b449ff177c0]::framework::engine::Results<rustc_mir_dataflow[99c12b449ff177c0]::impls::EverInitializedPlaces>>, rustc_borrowck[873188a397b846b5]::MirBorrowckCtxt>
                               at /Users/ekuber/workspace/rust/compiler/rustc_mir_dataflow/src/framework/direction.rs:468:13
  69:        0x1208deda5 - rustc_mir_dataflow[99c12b449ff177c0]::framework::visitor::visit_results::<rustc_borrowck[873188a397b846b5]::dataflow::BorrowckAnalyses<rustc_index[845bb2ac091a64a7]::bit_set::BitSet<rustc_borrowck[873188a397b846b5]::dataflow::BorrowIndex>, rustc_index[845bb2ac091a64a7]::bit_set::ChunkedBitSet<rustc_mir_dataflow[99c12b449ff177c0]::move_paths::MovePathIndex>, rustc_index[845bb2ac091a64a7]::bit_set::ChunkedBitSet<rustc_mir_dataflow[99c12b449ff177c0]::move_paths::InitIndex>>, rustc_borrowck[873188a397b846b5]::dataflow::BorrowckAnalyses<rustc_mir_dataflow[99c12b449ff177c0]::framework::engine::Results<rustc_borrowck[873188a397b846b5]::dataflow::Borrows>, rustc_mir_dataflow[99c12b449ff177c0]::framework::engine::Results<rustc_mir_dataflow[99c12b449ff177c0]::impls::MaybeUninitializedPlaces>, rustc_mir_dataflow[99c12b449ff177c0]::framework::engine::Results<rustc_mir_dataflow[99c12b449ff177c0]::impls::EverInitializedPlaces>>, core[3d5a77c8bca55160]::iter::adapters::map::Map<rustc_middle[73d8f7044761bf4e]::mir::traversal::ReversePostorderIter, rustc_borrowck[873188a397b846b5]::do_mir_borrowck::{closure#2}>, rustc_borrowck[873188a397b846b5]::MirBorrowckCtxt>
                               at /Users/ekuber/workspace/rust/compiler/rustc_mir_dataflow/src/framework/visitor.rs:25:9
  70:        0x1209b66ec - rustc_borrowck[873188a397b846b5]::do_mir_borrowck
                               at /Users/ekuber/workspace/rust/compiler/rustc_borrowck/src/lib.rs:390:5
  71:        0x1209a48db - rustc_borrowck[873188a397b846b5]::mir_borrowck
                               at /Users/ekuber/workspace/rust/compiler/rustc_borrowck/src/lib.rs:152:27
  72:        0x12097a413 - rustc_borrowck[873188a397b846b5]::provide::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_borrowck/src/lib.rs:117:17
  73:        0x12097a413 - <rustc_borrowck[873188a397b846b5]::provide::{closure#0} as core[3d5a77c8bca55160]::ops::function::FnOnce<(rustc_middle[73d8f7044761bf4e]::ty::context::TyCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId)>>::call_once
                               at /Users/ekuber/workspace/rust/library/core/src/ops/function.rs:422:5
  74:        0x1211430ee - <rustc_query_system[1d6662ecf82d368]::query::config::QueryVTable<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>>::compute
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_system/src/query/config.rs:66:9
  75:        0x1211430ee - rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job::<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_system/src/query/plumbing.rs:435:13
  76:        0x1211430ee - stacker[764ab632ce0b4f25]::maybe_grow::<&rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}>
                               at /Users/ekuber/.cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.15/src/lib.rs:55:9
  77:        0x1211430ee - rustc_data_structures[ecb593141c7f3dc9]::stack::ensure_sufficient_stack::<&rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}>
                               at /Users/ekuber/workspace/rust/compiler/rustc_data_structures/src/stack.rs:17:5
  78:        0x1211430ee - <rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query::<&rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}>::{closure#0}::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_impl/src/plumbing.rs:128:17
  79:        0x1211430ee - rustc_middle[73d8f7044761bf4e]::ty::context::tls::enter_context::<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<&rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}>::{closure#0}::{closure#0}, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:1980:50
  80:        0x1211430ee - rustc_middle[73d8f7044761bf4e]::ty::context::tls::set_tlv::<rustc_middle[73d8f7044761bf4e]::ty::context::tls::enter_context<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<&rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}>::{closure#0}::{closure#0}, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:1964:9
  81:        0x1211430ee - rustc_middle[73d8f7044761bf4e]::ty::context::tls::enter_context::<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<&rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}>::{closure#0}::{closure#0}, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:1980:9
  82:        0x1211430ee - <rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query::<&rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_impl/src/plumbing.rs:127:13
  83:        0x1211430ee - rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_related_context::<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<&rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}>::{closure#0}, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:2024:13
  84:        0x1211430ee - rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_context::<rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_related_context<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<&rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}>::{closure#0}, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:2008:40
  85:        0x1211430ee - rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_context_opt::<rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_context<rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_related_context<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<&rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}>::{closure#0}, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:1997:22
  86:        0x1211430ee - rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_context::<rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_related_context<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<&rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}>::{closure#0}, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:2008:9
  87:        0x1211430ee - rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_related_context::<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<&rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}>::{closure#0}, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:2021:9
  88:        0x1211430ee - <rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query::<&rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>::{closure#0}>
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_impl/src/plumbing.rs:112:9
  89:        0x1211430ee - rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job::<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_system/src/query/plumbing.rs:434:22
  90:        0x1211430ee - rustc_query_system[1d6662ecf82d368]::query::plumbing::try_execute_query::<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_query_system[1d6662ecf82d368]::query::caches::VecCache<rustc_span[8c1167581135dff5]::def_id::LocalDefId, &rustc_middle[73d8f7044761bf4e]::mir::query::BorrowCheckResult>>
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_system/src/query/plumbing.rs:379:44
  91:        0x1211b9ea9 - rustc_query_system[1d6662ecf82d368]::query::plumbing::get_query::<rustc_query_impl[8ec55ab5065c0b69]::queries::mir_borrowck, rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt>
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_system/src/query/plumbing.rs:760:36
  92:        0x120f135e8 - <rustc_query_impl[8ec55ab5065c0b69]::Queries as rustc_middle[73d8f7044761bf4e]::ty::query::QueryEngine>::mir_borrowck::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_impl/src/plumbing.rs:721:17
  93:        0x120f135e8 - <rustc_query_impl[8ec55ab5065c0b69]::Queries as rustc_middle[73d8f7044761bf4e]::ty::query::QueryEngine>::mir_borrowck
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_impl/src/lib.rs:45:1
  94:        0x11d5f14e4 - <rustc_middle[73d8f7044761bf4e]::ty::query::TyCtxtEnsure>::mir_borrowck
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/query.rs:234:17
  95:        0x11d5f14e4 - rustc_interface[cdce98f6f512c0f4]::passes::analysis::{closure#2}::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_interface/src/passes.rs:910:44
  96:        0x11d5f14e4 - <rustc_middle[73d8f7044761bf4e]::hir::map::Map>::par_body_owners::<rustc_interface[cdce98f6f512c0f4]::passes::analysis::{closure#2}::{closure#0}>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/hir/map/mod.rs:494:82
  97:        0x11d5f14e4 - rustc_data_structures[ecb593141c7f3dc9]::sync::par_for_each_in::<&[rustc_span[8c1167581135dff5]::def_id::LocalDefId], <rustc_middle[73d8f7044761bf4e]::hir::map::Map>::par_body_owners<rustc_interface[cdce98f6f512c0f4]::passes::analysis::{closure#2}::{closure#0}>::{closure#0}>::{closure#0}::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_data_structures/src/sync.rs:152:66
  98:        0x11d5f14e4 - <core[3d5a77c8bca55160]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[ecb593141c7f3dc9]::sync::par_for_each_in<&[rustc_span[8c1167581135dff5]::def_id::LocalDefId], <rustc_middle[73d8f7044761bf4e]::hir::map::Map>::par_body_owners<rustc_interface[cdce98f6f512c0f4]::passes::analysis::{closure#2}::{closure#0}>::{closure#0}>::{closure#0}::{closure#0}> as core[3d5a77c8bca55160]::ops::function::FnOnce<()>>::call_once
                               at /Users/ekuber/workspace/rust/library/core/src/panic/unwind_safe.rs:271:9
  99:        0x11d656ab9 - std[b33a454d9047b58c]::panicking::try::do_call::<core[3d5a77c8bca55160]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[ecb593141c7f3dc9]::sync::par_for_each_in<&[rustc_span[8c1167581135dff5]::def_id::LocalDefId], <rustc_middle[73d8f7044761bf4e]::hir::map::Map>::par_body_owners<rustc_interface[cdce98f6f512c0f4]::passes::analysis::{closure#2}::{closure#0}>::{closure#0}>::{closure#0}::{closure#0}>, ()>
                               at /Users/ekuber/workspace/rust/library/std/src/panicking.rs:483:40
 100:        0x11d656ab9 - std[b33a454d9047b58c]::panicking::try::<(), core[3d5a77c8bca55160]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[ecb593141c7f3dc9]::sync::par_for_each_in<&[rustc_span[8c1167581135dff5]::def_id::LocalDefId], <rustc_middle[73d8f7044761bf4e]::hir::map::Map>::par_body_owners<rustc_interface[cdce98f6f512c0f4]::passes::analysis::{closure#2}::{closure#0}>::{closure#0}>::{closure#0}::{closure#0}>>
                               at /Users/ekuber/workspace/rust/library/std/src/panicking.rs:447:19
 101:        0x11d5ed693 - std[b33a454d9047b58c]::panic::catch_unwind::<core[3d5a77c8bca55160]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[ecb593141c7f3dc9]::sync::par_for_each_in<&[rustc_span[8c1167581135dff5]::def_id::LocalDefId], <rustc_middle[73d8f7044761bf4e]::hir::map::Map>::par_body_owners<rustc_interface[cdce98f6f512c0f4]::passes::analysis::{closure#2}::{closure#0}>::{closure#0}>::{closure#0}::{closure#0}>, ()>
                               at /Users/ekuber/workspace/rust/library/std/src/panic.rs:137:14
 102:        0x11d5ed693 - rustc_data_structures[ecb593141c7f3dc9]::sync::par_for_each_in::<&[rustc_span[8c1167581135dff5]::def_id::LocalDefId], <rustc_middle[73d8f7044761bf4e]::hir::map::Map>::par_body_owners<rustc_interface[cdce98f6f512c0f4]::passes::analysis::{closure#2}::{closure#0}>::{closure#0}>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_data_structures/src/sync.rs:152:33
 103:        0x11d5ed693 - <core[3d5a77c8bca55160]::slice::iter::Iter<rustc_span[8c1167581135dff5]::def_id::LocalDefId> as core[3d5a77c8bca55160]::iter::traits::iterator::Iterator>::for_each::<rustc_data_structures[ecb593141c7f3dc9]::sync::par_for_each_in<&[rustc_span[8c1167581135dff5]::def_id::LocalDefId], <rustc_middle[73d8f7044761bf4e]::hir::map::Map>::par_body_owners<rustc_interface[cdce98f6f512c0f4]::passes::analysis::{closure#2}::{closure#0}>::{closure#0}>::{closure#0}>
                               at /Users/ekuber/workspace/rust/library/core/src/slice/iter/macros.rs:211:21
 104:        0x11d5ed693 - rustc_data_structures[ecb593141c7f3dc9]::sync::par_for_each_in::<&[rustc_span[8c1167581135dff5]::def_id::LocalDefId], <rustc_middle[73d8f7044761bf4e]::hir::map::Map>::par_body_owners<rustc_interface[cdce98f6f512c0f4]::passes::analysis::{closure#2}::{closure#0}>::{closure#0}>
                               at /Users/ekuber/workspace/rust/compiler/rustc_data_structures/src/sync.rs:151:13
 105:        0x11d5f45b0 - <rustc_middle[73d8f7044761bf4e]::hir::map::Map>::par_body_owners::<rustc_interface[cdce98f6f512c0f4]::passes::analysis::{closure#2}::{closure#0}>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/hir/map/mod.rs:494:9
 106:        0x11d5f45b0 - rustc_interface[cdce98f6f512c0f4]::passes::analysis::{closure#2}
                               at /Users/ekuber/workspace/rust/compiler/rustc_interface/src/passes.rs:910:9
 107:        0x11d5f45b0 - <rustc_data_structures[ecb593141c7f3dc9]::profiling::VerboseTimingGuard>::run::<(), rustc_interface[cdce98f6f512c0f4]::passes::analysis::{closure#2}>
                               at /Users/ekuber/workspace/rust/compiler/rustc_data_structures/src/profiling.rs:726:9
 108:        0x11d5f45b0 - <rustc_session[fe290d066b76030d]::session::Session>::time::<(), rustc_interface[cdce98f6f512c0f4]::passes::analysis::{closure#2}>
                               at /Users/ekuber/workspace/rust/compiler/rustc_session/src/utils.rs:10:9
 109:        0x11d5e7363 - rustc_interface[cdce98f6f512c0f4]::passes::analysis
                               at /Users/ekuber/workspace/rust/compiler/rustc_interface/src/passes.rs:909:5
 110:        0x12110f534 - <rustc_query_system[1d6662ecf82d368]::query::config::QueryVTable<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>>::compute
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_system/src/query/config.rs:66:9
 111:        0x12110f534 - rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job::<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_system/src/query/plumbing.rs:435:13
 112:        0x12110f534 - stacker[764ab632ce0b4f25]::maybe_grow::<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}>
                               at /Users/ekuber/.cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.15/src/lib.rs:55:9
 113:        0x12110f534 - rustc_data_structures[ecb593141c7f3dc9]::stack::ensure_sufficient_stack::<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}>
                               at /Users/ekuber/workspace/rust/compiler/rustc_data_structures/src/stack.rs:17:5
 114:        0x12110f534 - <rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query::<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}>::{closure#0}::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_impl/src/plumbing.rs:128:17
 115:        0x12110f534 - rustc_middle[73d8f7044761bf4e]::ty::context::tls::enter_context::<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}>::{closure#0}::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:1980:50
 116:        0x12110f534 - rustc_middle[73d8f7044761bf4e]::ty::context::tls::set_tlv::<rustc_middle[73d8f7044761bf4e]::ty::context::tls::enter_context<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}>::{closure#0}::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:1964:9
 117:        0x12110f534 - rustc_middle[73d8f7044761bf4e]::ty::context::tls::enter_context::<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}>::{closure#0}::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:1980:9
 118:        0x12110f534 - <rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query::<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_impl/src/plumbing.rs:127:13
 119:        0x12110f534 - rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_related_context::<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:2024:13
 120:        0x12110f534 - rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_context::<rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_related_context<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:2008:40
 121:        0x12110f534 - rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_context_opt::<rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_context<rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_related_context<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:1997:22
 122:        0x12110f534 - rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_context::<rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_related_context<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:2008:9
 123:        0x12110f534 - rustc_middle[73d8f7044761bf4e]::ty::context::tls::with_related_context::<<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:2021:9
 124:        0x12110f534 - <rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt as rustc_query_system[1d6662ecf82d368]::query::QueryContext>::start_query::<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}>
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_impl/src/plumbing.rs:112:9
 125:        0x12110f534 - rustc_query_system[1d6662ecf82d368]::query::plumbing::execute_job::<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, (), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_system/src/query/plumbing.rs:434:22
 126:        0x12110f534 - rustc_query_system[1d6662ecf82d368]::query::plumbing::try_execute_query::<rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt, rustc_query_system[1d6662ecf82d368]::query::caches::DefaultCache<(), core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>>
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_system/src/query/plumbing.rs:379:44
 127:        0x1212148ed - rustc_query_system[1d6662ecf82d368]::query::plumbing::get_query::<rustc_query_impl[8ec55ab5065c0b69]::queries::analysis, rustc_query_impl[8ec55ab5065c0b69]::plumbing::QueryCtxt>
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_system/src/query/plumbing.rs:760:36
 128:        0x120ef2b20 - <rustc_query_impl[8ec55ab5065c0b69]::Queries as rustc_middle[73d8f7044761bf4e]::ty::query::QueryEngine>::analysis::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_impl/src/plumbing.rs:721:17
 129:        0x120ef2b20 - <rustc_query_impl[8ec55ab5065c0b69]::Queries as rustc_middle[73d8f7044761bf4e]::ty::query::QueryEngine>::analysis
                               at /Users/ekuber/workspace/rust/compiler/rustc_query_impl/src/lib.rs:45:1
 130:        0x11d518c89 - <rustc_middle[73d8f7044761bf4e]::ty::query::TyCtxtAt>::analysis
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/query.rs:263:17
 131:        0x11d518c89 - <rustc_middle[73d8f7044761bf4e]::ty::context::TyCtxt>::analysis
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/query.rs:244:17
 132:        0x11d518c89 - rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}::{closure#2}::{closure#2}
                               at /Users/ekuber/workspace/rust/compiler/rustc_driver/src/lib.rs:375:30
 133:        0x11d518c89 - <rustc_interface[cdce98f6f512c0f4]::passes::QueryContext>::enter::<rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}::{closure#2}::{closure#2}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_interface/src/passes.rs:765:42
 134:        0x11d518c89 - rustc_middle[73d8f7044761bf4e]::ty::context::tls::enter_context::<<rustc_interface[cdce98f6f512c0f4]::passes::QueryContext>::enter<rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}::{closure#2}::{closure#2}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:1980:50
 135:        0x11d518c89 - rustc_middle[73d8f7044761bf4e]::ty::context::tls::set_tlv::<rustc_middle[73d8f7044761bf4e]::ty::context::tls::enter_context<<rustc_interface[cdce98f6f512c0f4]::passes::QueryContext>::enter<rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}::{closure#2}::{closure#2}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:1964:9
 136:        0x11d518c89 - rustc_middle[73d8f7044761bf4e]::ty::context::tls::enter_context::<<rustc_interface[cdce98f6f512c0f4]::passes::QueryContext>::enter<rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}::{closure#2}::{closure#2}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>
                               at /Users/ekuber/workspace/rust/compiler/rustc_middle/src/ty/context.rs:1980:9
 137:        0x11d518c89 - <rustc_interface[cdce98f6f512c0f4]::passes::QueryContext>::enter::<rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}::{closure#2}::{closure#2}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>
                               at /Users/ekuber/workspace/rust/compiler/rustc_interface/src/passes.rs:765:9
 138:        0x11d52d1e5 - rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}::{closure#2}
                               at /Users/ekuber/workspace/rust/compiler/rustc_driver/src/lib.rs:374:13
 139:        0x11d52d1e5 - <rustc_interface[cdce98f6f512c0f4]::interface::Compiler>::enter::<rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}::{closure#2}, core[3d5a77c8bca55160]::result::Result<core[3d5a77c8bca55160]::option::Option<rustc_interface[cdce98f6f512c0f4]::queries::Linker>, rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>
                               at /Users/ekuber/workspace/rust/compiler/rustc_interface/src/queries.rs:380:19
 140:        0x11d548301 - rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}
                               at /Users/ekuber/workspace/rust/compiler/rustc_driver/src/lib.rs:306:22
 141:        0x11d548301 - rustc_interface[cdce98f6f512c0f4]::interface::run_compiler::<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}>::{closure#0}::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_interface/src/interface.rs:327:21
 142:        0x11d548301 - rustc_span[8c1167581135dff5]::with_source_map::<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_interface[cdce98f6f512c0f4]::interface::run_compiler<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
                               at /Users/ekuber/workspace/rust/compiler/rustc_span/src/lib.rs:1016:5
 143:        0x11d54f9a1 - rustc_interface[cdce98f6f512c0f4]::interface::run_compiler::<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}>::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_interface/src/interface.rs:321:13
 144:        0x11d54f9a1 - <scoped_tls[c5a274eab3fc2fb4]::ScopedKey<rustc_span[8c1167581135dff5]::SessionGlobals>>::set::<rustc_interface[cdce98f6f512c0f4]::interface::run_compiler<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>
                               at /Users/ekuber/.cargo/registry/src/github.com-1ecc6299db9ec823/scoped-tls-1.0.0/src/lib.rs:137:9
 145:        0x11d4c35a0 - rustc_span[8c1167581135dff5]::create_session_globals_then::<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_interface[cdce98f6f512c0f4]::interface::run_compiler<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}>::{closure#0}>
                               at /Users/ekuber/workspace/rust/compiler/rustc_span/src/lib.rs:111:5
 146:        0x11d4c35a0 - rustc_interface[cdce98f6f512c0f4]::util::run_in_thread_pool_with_globals::<rustc_interface[cdce98f6f512c0f4]::interface::run_compiler<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}::{closure#0}
                               at /Users/ekuber/workspace/rust/compiler/rustc_interface/src/util.rs:145:38
 147:        0x11d4c35a0 - std[b33a454d9047b58c]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[cdce98f6f512c0f4]::util::run_in_thread_pool_with_globals<rustc_interface[cdce98f6f512c0f4]::interface::run_compiler<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>
                               at /Users/ekuber/workspace/rust/library/std/src/sys_common/backtrace.rs:121:18
 148:        0x11d5465aa - <std[b33a454d9047b58c]::thread::Builder>::spawn_unchecked_::<rustc_interface[cdce98f6f512c0f4]::util::run_in_thread_pool_with_globals<rustc_interface[cdce98f6f512c0f4]::interface::run_compiler<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#1}::{closure#0}
                               at /Users/ekuber/workspace/rust/library/std/src/thread/mod.rs:550:17
 149:        0x11d5465aa - <core[3d5a77c8bca55160]::panic::unwind_safe::AssertUnwindSafe<<std[b33a454d9047b58c]::thread::Builder>::spawn_unchecked_<rustc_interface[cdce98f6f512c0f4]::util::run_in_thread_pool_with_globals<rustc_interface[cdce98f6f512c0f4]::interface::run_compiler<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#1}::{closure#0}> as core[3d5a77c8bca55160]::ops::function::FnOnce<()>>::call_once
                               at /Users/ekuber/workspace/rust/library/core/src/panic/unwind_safe.rs:271:9
 150:        0x11d5465aa - std[b33a454d9047b58c]::panicking::try::do_call::<core[3d5a77c8bca55160]::panic::unwind_safe::AssertUnwindSafe<<std[b33a454d9047b58c]::thread::Builder>::spawn_unchecked_<rustc_interface[cdce98f6f512c0f4]::util::run_in_thread_pool_with_globals<rustc_interface[cdce98f6f512c0f4]::interface::run_compiler<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#1}::{closure#0}>, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>
                               at /Users/ekuber/workspace/rust/library/std/src/panicking.rs:483:40
 151:        0x11d5465aa - std[b33a454d9047b58c]::panicking::try::<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, core[3d5a77c8bca55160]::panic::unwind_safe::AssertUnwindSafe<<std[b33a454d9047b58c]::thread::Builder>::spawn_unchecked_<rustc_interface[cdce98f6f512c0f4]::util::run_in_thread_pool_with_globals<rustc_interface[cdce98f6f512c0f4]::interface::run_compiler<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#1}::{closure#0}>>
                               at /Users/ekuber/workspace/rust/library/std/src/panicking.rs:447:19
 152:        0x11d51f4c0 - std[b33a454d9047b58c]::panic::catch_unwind::<core[3d5a77c8bca55160]::panic::unwind_safe::AssertUnwindSafe<<std[b33a454d9047b58c]::thread::Builder>::spawn_unchecked_<rustc_interface[cdce98f6f512c0f4]::util::run_in_thread_pool_with_globals<rustc_interface[cdce98f6f512c0f4]::interface::run_compiler<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#1}::{closure#0}>, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>
                               at /Users/ekuber/workspace/rust/library/std/src/panic.rs:137:14
 153:        0x11d51f4c0 - <std[b33a454d9047b58c]::thread::Builder>::spawn_unchecked_::<rustc_interface[cdce98f6f512c0f4]::util::run_in_thread_pool_with_globals<rustc_interface[cdce98f6f512c0f4]::interface::run_compiler<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#1}
                               at /Users/ekuber/workspace/rust/library/std/src/thread/mod.rs:549:30
 154:        0x11d51f4c0 - <<std[b33a454d9047b58c]::thread::Builder>::spawn_unchecked_<rustc_interface[cdce98f6f512c0f4]::util::run_in_thread_pool_with_globals<rustc_interface[cdce98f6f512c0f4]::interface::run_compiler<core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>, rustc_driver[c86f44892fce95b7]::run_compiler::{closure#1}>::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[3d5a77c8bca55160]::result::Result<(), rustc_errors[ca653ba93ce493de]::ErrorGuaranteed>>::{closure#1} as core[3d5a77c8bca55160]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
                               at /Users/ekuber/workspace/rust/library/core/src/ops/function.rs:422:5
 155:        0x10de57e9b - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h9b6dd92b5e82c886
                               at /Users/ekuber/workspace/rust/library/alloc/src/boxed.rs:1990:9
 156:        0x10de57e9b - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::heff55171a0754b3c
                               at /Users/ekuber/workspace/rust/library/alloc/src/boxed.rs:1990:9
 157:        0x10de4c0aa - std::sys::unix::thread::Thread::new::thread_start::hcf1ea46076ac3acb
                               at /Users/ekuber/workspace/rust/library/std/src/sys/unix/thread.rs:108:17
 158:     0x7ff80c6544e1 - __pthread_start

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.68.0-dev running on x86_64-apple-darwin

note: compiler flags: -Z threads=1 -C codegen-units=1 -Z ui-testing -Z simulate-remapped-rust-src-base=/rustc/FAKE_PREFIX -Z translate-remapped-path-to-local-path=no -Z deduplicate-diagnostics=no -C strip=debuginfo -C prefer-dynamic -C rpath -C debuginfo=0

query stack during panic:
#0 [mir_borrowck] borrow-checking `main`
#1 [analysis] running analysis passes on this crate
end of query stack
------------------------------------------
```

Reply:
How peculiar -- that doesn't look like an erased region.. it looks like a region variable being passed to `can_eq`.  I checked out your branch and changed the `freshen` to `erase_regions` (and removed the one I mentioned below) and can't reproduce this ICE.

Here's my change on top of your `pin-mut-reborrow` branch:

```diff
--- a/compiler/rustc_borrowck/src/diagnostics/mod.rs
+++ b/compiler/rustc_borrowck/src/diagnostics/mod.rs
@@ -1135,17 +1135,16 @@ fn explain_captures(
                             ),
                         );
                         let infcx = tcx.infer_ctxt().build();
-                        let ty = infcx.freshen(moved_place.ty(self.body, tcx).ty);
+                        let ty = tcx.erase_regions(moved_place.ty(self.body, tcx).ty);
                         if let ty::Adt(def, substs) = ty.kind()
                             && Some(def.did()) == tcx.lang_items().pin_type()
                             && let ty::Ref(_, _, hir::Mutability::Mut) = substs.type_at(0).kind()
-                            && let self_ty = infcx.freshen(
+                            && let self_ty =
                                 infcx.replace_bound_vars_with_fresh_vars(
                                     fn_call_span,
                                     LateBoundRegionConversionTime::FnCall,
                                     tcx.fn_sig(method_did).input(0),
                                 )
-                            )
                             && infcx.can_eq(self.param_env, ty, self_ty).is_ok()
                         {
                             err.span_suggestion_verbose(
```

Reply:
I could have sworn that I tried exactly this and it didn't work.